src/sunheuristics/default/sunheuristics_default.c

```diff
@@ -0,0 +1,488 @@
+/* -----------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ * -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2023, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * This is the implementation file for the SUNHeuristics_Default
+ * module.
+ * -----------------------------------------------------------------*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sunheuristics/sunheuristics_default.h>
+#include <sundials/sundials_math.h>
+
+
+/* ---------------
+ * Macro accessors
+ * --------------- */
+
+#define SH_CONTENT(H)     ( (SUNHeuristicsContent_Default)(H->content) )
+#define SH_HMAX_INV(H)    ( SH_CONTENT(H)->hmax_inv )
+#define SH_HMIN(H)        ( SH_CONTENT(H)->hmin )
+#define SH_ETAMAX(H)      ( SH_CONTENT(H)->etamax )
+#define SH_ETAMX1(H)      ( SH_CONTENT(H)->etamx1 )
+#define SH_ETAMXF(H)      ( SH_CONTENT(H)->etamxf )
+#define SH_ETAMIN(H)      ( SH_CONTENT(H)->etamin )
+#define SH_SMALL_NEF(H)   ( SH_CONTENT(H)->small_nef )
+#define SH_ETACF(H)       ( SH_CONTENT(H)->etacf )
+#define SH_EXPSTAB(H)     ( SH_CONTENT(H)->expstab )
+#define SH_ESTAB_DATA(H)  ( SH_CONTENT(H)->estab_data )
+#define SH_CFL(H)         ( SH_CONTENT(H)->cfl )
+#define SH_SAFETY(H)      ( SH_CONTENT(H)->safety )
+#define SH_GROWTH(H)      ( SH_CONTENT(H)->growth )
+#define SH_LBOUND(H)      ( SH_CONTENT(H)->lbound )
+#define SH_UBOUND(H)      ( SH_CONTENT(H)->ubound )
+#define SH_NST_ACC(H)     ( SH_CONTENT(H)->nst_acc )
+#define SH_NST_EXP(H)     ( SH_CONTENT(H)->nst_exp )
+
+/* ------------------
+ * Default parameters
+ * ------------------ */
+
+#define DEFAULT_CFLFAC     RCONST(0.5)
+#define DEFAULT_SAFETY     RCONST(0.96)
```

Comment:
With the fix to the controller exponent, I'd suggest setting this to 0.9 which is a pretty conventional value. Then the bias should be set to 1 along with that.